pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -390,6 +421,20 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	}
 }
 
+func lookupDestIPID(destIPStr string) (SecID ipcache.Identity, exists bool, err error) {
+	maxPrefixLen := 32
+	if ip := net.ParseIP(destIPStr); ip.To4() == nil {
+		maxPrefixLen = 128
+	}
+
+	for i := maxPrefixLen; exists != true && i >= 0; i-- {
```

Comment:
We actually store the currently-used prefixes in the daemon somewhere (`prefixCounter`) if you want to go that way; it's probably simpler than importing a full LPM into userspace, particularly for a backport.